src/index.ts

```diff
@@ -1,65 +1,71 @@
-import cli from "cli-ux";
+import { Args, Command, Flags, ux } from "@oclif/core";
 
 import * as parser from "./helpers/parser";
 import * as tsReader from "./helpers/tsReader";
 import * as paths from "./helpers/paths";
 import * as formatter from "./helpers/formatter";
 import * as generator from "./helpers/generator";
+import * as cli from "./helpers/cli";
 
 class MongooseTsgen extends Command {
+  static id = ".";
+
   static description =
     "Generate a Typescript file containing Mongoose Schema typings.\nSpecify the directory of your Mongoose model definitions using `MODEL_PATH`. If left blank, all sub-directories will be searched for `models/*.ts` (ignores `index.ts` files). Files found are expected to export a Mongoose model.";
 
   static flags = {
-    config: flags.string({
+    config: Flags.string({
       char: "c",
       description:
         "[default: ./] Path of `mtgen.config.json` or its root folder. CLI flag options will take precendence over settings in `mtgen.config.json`."
     }),
-    "dry-run": flags.boolean({
+    "dry-run": Flags.boolean({
       char: "d",
       description: "Print output rather than writing to file."
     }),
-    help: flags.help({ char: "h" }),
-    imports: flags.string({
+    help: cli.helpFlag({
+      char: "h"
+    }),
+    imports: Flags.string({
       char: "i",
       description:
         "Custom import statements to add to the output file. Useful if you use third-party types in your mongoose schema definitions. For multiple imports, specify this flag more than once.",
       multiple: true
     }),
-    "no-format": flags.boolean({
+    "no-format": Flags.boolean({
       description: "Disable formatting generated files with prettier."
     }),
-    output: flags.string({
+    output: Flags.string({
       char: "o",
       description:
         "[default: ./src/interfaces] Path of output file to write generated typings. If a folder path is passed, the generator will create a `mongoose.gen.ts` file in the specified folder."
     }),
-    project: flags.string({
+    project: Flags.string({
       char: "p",
       description: "[default: ./] Path of `tsconfig.json` or its root folder."
     }),
-    debug: flags.boolean({
+    debug: Flags.boolean({
       description: "Print debug information if anything isn't working"
     }),
-    "no-mongoose": flags.boolean({
+    "no-mongoose": Flags.boolean({
       description:
         "Don't generate types that reference mongoose (i.e. documents). Replace ObjectId with string."
     }),
-    "no-populate-overload": flags.boolean({
+    "no-populate-overload": Flags.boolean({
       description:
         "Disable augmenting mongoose with Query.populate overloads (the overloads narrow the return type of populated documents queries)."
     })
   };
 
   // path of mongoose models folder
-  static args = [{ name: "model_path" }];
+  static args = {
+    model_path: Args.string({ required: true })
+  };
```

Comment:
https://github.com/francescov1/mongoose-tsgen/pull/112#issuecomment-1444813255 that's likely from this change here - I guessed at `required`, but i'll remove that!

Reply:
Thanks!